tests/basic/async_dr_test.go

```diff
@@ -637,13 +639,274 @@ var _ = Describe("{StorkctlPerformFailoverFailbackeckEsClusterwide}", func() {
+		log.InfoD("upgrade volume driver and ensure everything is running fine")
+
+		taskNamePrefix := "asyncdr-upgradepx"
+		defaultNs := "kube-system"
+		migrationNamespaces, contexts := initialSetupApps(taskNamePrefix, false)
+		migNamespaces := strings.Join(migrationNamespaces, ",")
+		kubeConfigPathSrc, err := GetCustomClusterConfigPath(asyncdr.FirstCluster)
+		log.FailOnError(err, "Failed to get source configPath: %v", err)
+		kubeConfigPathDest, err := GetCustomClusterConfigPath(asyncdr.SecondCluster)
+		log.FailOnError(err, "Failed to get destination configPath: %v", err)
+		var migrationSchedName string
+		var schdPol *storkapi.SchedulePolicy
+		cpName := defaultClusterPairName + time.Now().Format("15h03m05s")
+		scpolName := "async-policy"
+		migrationInterval := 5
+		Step("Create Schedule Policy", func() {
+			schdPol, err = asyncdr.CreateSchedulePolicy(scpolName, migrationInterval)
+			log.FailOnError(err, "Failed to create schedule policy")
+		})
+
+		extraArgs := map[string]string{
+			"namespaces":           migNamespaces,
+			"kubeconfig":           kubeConfigPathSrc,
+			"schedule-policy-name": schdPol.Name,
+		}
+
+		Step("create clusterpair and start migration", func() {
+			log.InfoD("Creating clusterpair between first and second cluster")
+			err = ScheduleBidirectionalClusterPair(cpName, defaultNs, "", storkapi.BackupLocationType(defaultBackupLocation), defaultSecret, "async-dr", asyncdr.FirstCluster, asyncdr.SecondCluster)
+			log.FailOnError(err, "Failed creating bidirectional cluster pair")
+
+			log.InfoD("Start migration schedule and perform failover")
+			migrationSchedName = migrationSchedKey + time.Now().Format("15h03m05s")
+			createMigSchdAndValidateMigration(migrationSchedName, cpName, defaultNs, kubeConfigPathSrc, extraArgs)
+		})
+
+		storageNodes := node.GetStorageNodes()
+		numOfNodes := len(node.GetStorageDriverNodes())
+
+		//AddDrive is added to test to Vsphere Cloud drive upgrades when kvdb-device is part of storage in non-kvdb nodes
+		isCloudDrive, err := IsCloudDriveInitialised(storageNodes[0])
+		log.FailOnError(err, "Cloud drive installation failed")
+
+		if !isCloudDrive {
+			for _, storageNode := range storageNodes {
+				err := Inst().V.AddBlockDrives(&storageNode, nil)
+				if err != nil && strings.Contains(err.Error(), "no block drives available to add") {
+					continue
+				}
+				log.FailOnError(err, "Adding block drive(s) failed.")
+			}
+		}
+
+		err = hardSetConfig(kubeConfigPathDest)
+		log.FailOnError(err, "Switching context to destination cluster failed")
+		err = SetCustomKubeConfig(asyncdr.SecondCluster)
+		log.FailOnError(err, "Switching context to destination cluster failed")
+
+		err = Inst().S.RefreshNodeRegistry()
+		log.FailOnError(err, "Node registry refresh failed")
+		err = Inst().V.RefreshDriverEndpoints()
+		log.FailOnError(err, "Refresh Driver end points failed")
+		storageNodesDest := node.GetStorageNodes()
```

Comment:
Can we have map with key as cluster name and value as node array?

Reply:
Done!